include/envoy/server/access_log_config.h

```diff
@@ -31,7 +31,7 @@ class AccessLogInstanceFactory : public Config::TypedFactory {
    */
   virtual AccessLog::InstanceSharedPtr createAccessLogInstance(const Protobuf::Message& config,
                                                                AccessLog::FilterPtr&& filter,
-                                                               FactoryContext& context) PURE;
+                                                               ServerFactoryContext& context) PURE;
```

Comment:
Why did this need to change? I don't think we actually want this to be server context since access logs are created by filters?

Reply:
> Why did this need to change?

Because in order to call `AccessLog::AccessLogFactory::fromProto` I need a `FactoryContext`. Since the `FactoryContext` is constructed by listeners/filters I don't have access to it at this point. I realized though that the `ServerFactoryContext` should have most of information available.

I was waiting for the CI to pass before I manually verify these changes

Am I missing something here?

```
InitialImpl::InitialImpl(const envoy::config::bootstrap::v3::Bootstrap& bootstrap,
                         const Options& options, Instance& server)
    : enable_deprecated_v2_api_(options.bootstrapVersion() == 2u) {
  const auto& admin = bootstrap.admin();

  for (const auto& access_log : admin.access_log()) {
    AccessLog::InstanceSharedPtr current_access_log =
        AccessLog::AccessLogFactory::fromProto(access_log, server.serverFactoryContext());
    admin_.access_logs_.emplace_back(current_access_log);
  }

  if (!admin.access_log_path().empty()) {
    admin_.access_logs_.emplace_back(new Extensions::AccessLoggers::File::FileAccessLog(
        admin.access_log_path(), {},
        Formatter::SubstitutionFormatUtils::defaultSubstitutionFormatter(),
        server.accessLogManager()));
  }
  ```

Reply:
See my comment below.

Reply:
fixed in b8c57d538b2084b98a42d4ad5a92645c7f27b3d0